host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -1203,12 +1246,18 @@ static void stream_tx_xfer_submit(usb_transfer_t *out_xfer)
     uac_iface_t *iface = out_xfer->context;
     assert(iface);
 
+    // Prevent other threads from manipulating the ring buffer when calling the `uac_host_device_write` function.
+    xSemaphoreTake(iface->ringbuf_mutex, portMAX_DELAY);
```

Comment:
@TDA-2030 Sorry for not getting back to you sooner. Does this fix your issue? As far as I know, xRingbufferSend already has a Semaphore inside, and I don't see how this additional protection can play any other role

Reply:
After testing, the problem did indeed disappear after the modification. Without adding this `ringbuf_mutex`, this error would occasionally occur: https://github.com/espressif/esp-usb/blob/ef2dc82da2cc5027b2f0021bfea608d941043aac/host/class/uac/usb_host_uac/uac_host.c#L1256